### DISCUSSION TITLE
Does add_member_self_weight() actually works ?

### DISCUSSION THREAD

**Discussion by Terre-ronce** [JWock82/Pynite#193] [Q&A] [answered]
Hi,

Second time here, I have been using PyNite for about two months now and it has been working just great! Thanks to all the team for the great work.

Today my question is about the recently released version with add_member_self_weight(). I tried to use it but it does not work as I get the following error: `AttributeError: 'str' object has no attribute 'rho'`. This error is related to the fact that member.material only stores the name of the material of the member.

Also, by checking the source code I found in the FEModel3D() class an attribute Materials which is actualy a dictionnary and has the attribute rho so for instance if in the add_member_self_weight() method there was:

       # Step through each member in the model
        for member in self.Members.values():

            # Calculate the self weight of the member
            self_weight = factor*self.Materials[member.material].rho*member.A

insted of 

       # Step through each member in the model
        for member in self.Members.values():

            # Calculate the self weight of the member
            self_weight = factor*member.material.rho*member.A

I think it would work right. However I don't know if there is really an issue with PyNite code or if I am doing something wronf in my code.

Edit: I made the mentionned modification in my copy of PyNite, now there is no error anymore however, there is no load applied for some reason I cannot figure out. 

((( Also, I think the gravitationnal pull is missing (but I might be wrong i dont know what unit you use in the US, in france we use kg/m^3 for rho so the correct formula for the weight would be with our units:

 ` self_weight =  factor*self.Materials[member.material].rho*member.A*9.81
` 
to have a lineic load. Or maybe specify in the docstring of the method that the `factor` should include the gravitationnal pull. )))

Thanks in advance to whoever could help me with this!
Therence.

**Comment by JWock82** [comment, marked as answer]
Try it again with version 0.0.94. This issue should be fixed now.

**Reply by Terre-ronce** [reply]
Just did it and yes it is! thanks!!
